FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_Sockets.c

```diff
@@ -2271,12 +2275,15 @@ void vSocketWakeUpUser( FreeRTOS_Socket_t *pxSocket )
 	FreeRTOS_Socket_t *pxSocket = ( FreeRTOS_Socket_t* ) xClientSocket;
 	TickType_t xRemainingTime;
 	BaseType_t xTimed = pdFALSE;
-	BaseType_t xResult;
+	BaseType_t xResult = -pdFREERTOS_ERRNO_EINVAL;
 	TimeOut_t xTimeOut;
 
 		( void ) xAddressLength;
 
-		xResult = prvTCPConnectStart( pxSocket, pxAddress );
+		if( pxAddress != NULL )
```

Comment:
Why are we validating `pxAddress` but not `pxSocket` or `xAddressLength`?

Reply:
`pxSocket` is validated by `prvTCPConnectStart` which checks fo the socket being `NULL` or `FREERTOS_INVALID_SOCKET`.

`xAddressLength` is not used at all in this implementation. See a couple of lines above this comment: `( void ) xAddressLength;` is used to ignore the parameter `xAddressLength`.

Reply:
Then if `prvTCPConnectStart()` validates one parameter `pxSocket` why doesn't it validate both of them?  It is weird validating one parameter in each routine, why not just do the validations in one place?

Reply:
I agree with you on this front. `prvTCPConnectStart()` is used only in this particular function so it can be easily modified without any repercussions. I'll update that!
Thanks :)

Reply:
Done!